userspace/libscap/scap_savefile.c

```diff
@@ -541,15 +542,12 @@ static int32_t scap_write_proclist(scap_t *handle, scap_dumper_t *d)
 		return SCAP_SUCCESS;
 	}
 
-	//
-	// Exit immediately if the process list is empty
-	//
-	if(HASH_COUNT(handle->m_proclist.m_proclist) == 0)
+	scap_dumper_t *proclist_dumper = scap_write_proclist_begin(handle);
+	if(proclist_dumper == NULL)
 	{
-		return SCAP_SUCCESS;
+		return SCAP_FAILURE;
 	}

```

Comment:
this is a bit different: https://github.com/falcosecurity/libs/pull/888/files#diff-7b4e2af897ce4b8645f4f862762f68425d8394805c45c194255730e99078767dR717

probably needs to be fixed

Reply:
That change is not present in the `release/0.10.x` branch: https://github.com/falcosecurity/libs/blame/release/0.10.x/userspace/libscap/scap_savefile.c#L552
We need to keep it this way. Or we backport the commit that introduced that change into `release/0.10.x` branch.

Reply:
Do we miss the
```
	//
	// Exit immediately if the process list is empty
	//
	if(HASH_COUNT(handle->m_proclist.m_proclist) == 0)
	{
		return SCAP_SUCCESS;
	}
```
check now? It was present back then in 0.10.x.

Reply:
:+1: